app/page-partials/my-dashboard/IntegrationInfoTabs/MetricsPanel.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { Integration } from 'interfaces/Request';
+import { withTopAlert } from 'layout/TopAlert';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Tabs, Tab } from '@bcgov-sso/common-react-components';
+import startCase from 'lodash.startcase';
+import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, Text, Legend } from 'recharts';
+import { getMetrics } from '@app/services/grafana';
+import throttle from 'lodash.throttle';
+import moment from 'moment';
+import DateTimePicker from '@app/components/DateTimePicker';
+import { InfoMessage } from '@app/components/MessageBox';
+import { Link } from '@button-inc/bcgov-theme';
+
+export const DatePickerContainer = styled.div`
+  height: 100%;
+  display: flex;
+  align-items: center;
+  padding-right: 15px;
+  & > * {
+    margin-left: 15px;
+  }
+`;
+
+const Label = styled.label`
+  margin-bottom: 2px;
+`;
+
+const TopMargin = styled.div`
+  height: var(--field-top-spacing);
+`;
+
+const LeftTitle = styled.span`
+  color: #000;
+  font-size: 1.1rem;
+  font-weight: bold;
+`;
+
+const PaddedIcon = styled(FontAwesomeIcon)`
+  margin-right: 20px;
+`;
+
+const StyledP = styled.div`
+  margin-bottom: 5px;
+  display: flex;
+  align-items: center;
+`;
+
+const StyledHr = styled.hr`
+  background-color: black;
+`;
+
+interface Props {
+  integration: Integration;
+}
+
+const metricsStartDate = 'December 01, 2023';
+const MetricsPanel = ({ integration }: Props) => {
+  const [environment, setEnvironment] = useState('dev');
+  const environments = integration?.environments || [];
+  const [metrics, setMetrics] = useState<any>([]);
```

Comment:
If easy enough it would be nice to have a type for the metrics

Reply:
Thanks @jlangy I have updated the type